citesphere/src/test/java/edu/asu/diging/citesphere/core/service/oauth/impl/OAuthClientManagerTest.java

```diff
@@ -41,5 +46,21 @@ public void test_deleteClient_nullId() {
         managerToTest.deleteClient(clientId);
         Mockito.verify(clientRepo, Mockito.never()).deleteById(clientId);
     }
+    
+    @Test
+    public void test_getClientsDetails() {
+        List<String> clientList = new ArrayList<>();
+        clientList.add("Client1");
+        List<OAuthClient> clients = new ArrayList<>();
+        OAuthClient client = new OAuthClient();
+        clients.add(client);
+        Mockito.when(clientRepo.findAllById(clientList)).thenReturn(clients);
+        Assert.assertEquals(clients, managerToTest.getClientsDetails(clientList));
```

Comment:
what you really want to test is if the clients you get are the ones you requested (have the correct ids), right? Otherwise your test will fail even if the method would create a new list and copied all the client data into that list and return that (which would be ok) and it will succeed even if the method would modify the objects before returning and for example change their ids.